### ISSUE TITLE
Unable to open example_scenes.py SquareToCircle

### ISSUE THREAD

**Issue by zhangwn** [3b1b/manim#156]
Hi,
I'm getting the following error message
<img width="572" alt="capture" src="https://user-images.githubusercontent.com/28883499/37250442-4c30cc7c-24c2-11e8-8cba-ef1ead7f8c7d.PNG">
Can someone help me figure out what's wrong?

**Comment by jacobhelbig** [comment]
Are both extract_scene.py and example_scenes.py where they belong? And is the class SquareToCircle the way it should be?

**Comment by zhangwn** [comment]
They're in the manim-master folder. Should they not be?
I downloaded the files for using manim yesterday, and I followed the instruction on how to open a scene, but the error message appears.

**Comment by zimmermant** [comment]
I'm having the same issue too.

**Comment by jacobhelbig** [comment]
No, they should be in the root folder - Maybe something with your constants.py isn't correct, so it can't figure out what file to look for? Also: what is Anaconda2? It seems to be interfering with manim... (the last 2 error messages) that‘s probably it.

**Comment by zhangwn** [comment]
Anaconda2 is the Python distribution that I'm using. I have both Python 2 and Python 3 on my computer since Python 3 isn't backwards compatible. I use the command `py` for Python 2.7 and `python` for Python 3.6.

Anyways, how do I edit the constants.py so that it can look for the files in the manim-master folder? I don't want to clutter up my root directory.

**Comment by zhangwn** [comment]
Actually, I'm pretty sure that running the command `py extract_scene.py example_scenes.py -p` that it's able to find the extract_scene.py and example_scenes.py files just fine, since if I instead write `py fake_file.py -p` then I get a different error message.

**Comment by zhangwn** [comment]
One thing that I've tried that seems to do something (but doesn't resolve the issue completely) is replacing line 606 `self.writing_process = sp.Popen(command, stdin=sp.PIPE)` with `self.writing_process = sp.Popen(command, stdin=sp.PIPE, shell=True)` in the scene.py file.

**Comment by vors** [comment]
@zhangwn  What's the error with `shell=True`?

**Comment by zhangwn** [comment]
<img width="663" alt="capture" src="https://user-images.githubusercontent.com/28883499/37260449-3edcb8fc-2561-11e8-8474-c56e15bee6c0.PNG">

It's basically the same, but this time it at least succeeds in creating the SquareToCircle.mp4, although that video only shows a black screen.

**Comment by vors** [comment]
> It's basically the same

It's actually quite different: this second error comes just from the inability to open the mp4 file after it's written with `open` command (macOS specific).
So the real issue is the black screen...

**Comment by zhangwn** [comment]
Ok, so how would I fix this?

**Comment by zhangwn** [comment]
So, does that mean that the manim project wasn't intended to be compatible with Windows if it's relying on macOS specific commands? How would I fix the second error message?

**Comment by vors** [comment]
From my very limited experience in #162 it seems that some additional massaging needed for both Windows and Linux (you actually got further in Windows path then I did :D ). I don't think there are any fundamental problems since python is in most cases behave pretty reasonable in a cross-platform manner.

@3b1b can you shade some light on the current platforms where it should / known to work?

**Comment by iammosespaulr** [comment]
@3b1b I get the same error too ... Im in Ubuntu 17 btw ,
```
Writing to /home/cyborg/manim/videos/animations/example_scenes/1080p60/SquareToCircleTemp.mp4



Traceback (most recent call last):
  File "extract_scene.py", line 253, in main
    handle_scene(SceneClass(**scene_kwargs), **config)
  File "/home/cyborg/manim/scene/scene.py", line 65, in __init__
    self.open_movie_pipe()
  File "/home/cyborg/manim/scene/scene.py", line 606, in open_movie_pipe
    self.writing_process = sp.Popen(command, stdin=sp.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1025, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory



sh: 1: play: not found

```

**Comment by MedNait** [comment]
I tried installing manim on windows 10 and on ubuntu 16.04 without success, getting in the end, the apparently now famous, black screen issue. Different issues arise for each OS. For ubuntu, I was able to solve the following:

- Issue with the "open" command: using "xdg-open" command instead 
- Issue with the "play" command: installing SoX "sudo apt-get install sox" and its dependencies "sudo apt-get install libasound2-plugins libasound2-python libsox-fmt-all"

**Comment by MedNait** [comment]
I was able to get around the __black screen issue__ and other issues (open command for e.g.) and get `SquareToCircle` scene to work. My comment on issue #173 might be helpful.

__Disclaimer__: This worked for me on Windows.

**Comment by eulertour** [comment]
Closing, as this problem was solved by #202.
